db-migration.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+## COLORS
+BOLD=$(tput bold)
+RED=$(tput setaf 1)
+GREEN=$(tput setaf 2)
+YELLOW=$(tput setaf 3)
+RESET=$(tput sgr0)
+
+## VARS
+OUTPUT="/tmp/db-migration-$(date +"%s").dir" # where we save the dump, we use timestamp for uniqueness
+JOBS=$(($(nproc --all)-2)) # Number of cpu cores minus 2 to do parallel restore jobs
+SOURCE_URL="$1"
+TARGET_URL="$2"
+
+function show_help {
+    echo "${BOLD}Atlassian DB Migration Script${RESET}"
+    echo ""
+    echo "You must have postgresql-client and curl installed first."
+    echo ""
+    echo "Basic usage: ./$(basename $0) [source-db-url] [target-db-url]"
+    echo "Example: ./$(basename $0) postgres://postgres:postgres@localhost:5432/source-db postgres://postgres:postgres@localhost:5432/target-db"
+    echo ""
+    echo "${BOLD}Options${RESET}:"
+    echo " ${BOLD}-h${RESET}: Help - Show me this helpful message."
+}
+
+# Check if we can connect to DB
+function check_db {
+  if [ -z "$1" ]
+  then
+    echo "${RED}$2 database URL must be specified.${RESET}"
+    exit 1
+  fi
+  echo "${YELLOW}Checking connection to $2 database...${RESET}"
+  pg_isready --timeout 60 --dbname "$1"
+  if [ "$?" == "0" ]
+  then
+    echo "${GREEN}Successful connection to $2 database!${RESET}"
+  else
+    echo "${RED}Cannot connect to $2 database with URL '$1'. Exiting.${RESET}"
+    exit 1
+  fi
+}
+
+# Gather options from flags.
+while getopts "h:help:?" opt; do
+    case "$opt" in
+    h)
+        show_help
+        exit 0
+        ;;
+    help)
+        show_help
+        exit 0
+        ;;
+    \?)
+        show_help
+        exit 0
+        ;;
+    esac
+done
+
+# Check to see if postgresql-client is installed on the system
+if ! [ -x "$(command -v pg_isready)" ]; then
+  echo "${RED}postgresql-client is not installed, please install it first.${RESET}"
+  exit 1
+fi
+
+# VPN Connection check - URL is only accessible internally in VPN
+echo "${YELLOW}Checking VPN connection...${RESET}"
+curl -s -m 5 https://statlas.prod.atl-paas.net/ > /dev/null
+if ! [ "$?" == "0" ]
+then
+  echo "${RED}Not connected to Atlassian VPN.  Please connect to it first before continuing.${RESET}"
+  exit 1
+fi
+echo "${GREEN}Connected to Atlassian VPN.${RESET}"
+
+# Enter source url if it wasn't added as a parameter
+if [ -z "$SOURCE_URL" ]
+then
+  read -p "Enter the Source Database URL: " SOURCE_URL
+fi
+check_db $SOURCE_URL "Source"
+
+# Enter target url if it wasn't added as a parameter
+if [ -z "$TARGET_URL" ]
+then
+  read -p "Enter the Target Database URL: " TARGET_URL
+fi
+check_db $TARGET_URL "Target"
+
+echo "Databases ready for migration from '$SOURCE_URL' to '$TARGET_URL'."
+
+# Confirm that the user wants to continue
+read -p "${YELLOW}${BOLD}This will overwrite Target Database, are you sure you want to continue? [y/N] ${RESET}" PROMPT
+if ! [ "${PROMPT^^}" == "Y" ]
```

Comment:
Is bash case sensitive? The line above suggests user to type "y" but we're checking for "Y".

Reply:
it is, but the `^^` in `"${PROMPT^^}"` turns it to uppercase instead of having a check for both `y` and `Y` :)

Reply:
Oh sweet!